executor/src/witgen/query_processor.rs

```diff
@@ -139,30 +139,24 @@ where
         next: bool,
         rows: &RowPair<T>,
     ) -> Result<T, IncompleteCause<&'a AlgebraicReference>> {
-        if poly.index.is_none() {
-            let poly_id = poly.poly_id.unwrap();
-            match poly_id.ptype {
-                PolynomialType::Committed | PolynomialType::Intermediate => {
-                    let poly_ref = AlgebraicReference {
-                        name: poly.name.clone(),
-                        poly_id,
-                        index: poly.index,
-                        next,
-                    };
-                    Ok(rows
-                        .get_value(&poly_ref)
-                        .ok_or(IncompleteCause::DataNotYetAvailable)?)
-                }
-                PolynomialType::Constant => {
-                    let values = self.fixed_data.fixed_cols[&poly_id].values;
-                    let row = rows.current_row_index as usize + next as usize;
-                    Ok(values[row % values.len()])
-                }
+        let poly_id = poly.poly_id.unwrap();
+        match poly_id.ptype {
+            PolynomialType::Committed | PolynomialType::Intermediate => {
+                let poly_ref = AlgebraicReference {
+                    name: poly.name.clone(),
+                    poly_id,
+                    index: None,
+                    next,
+                };
+                Ok(rows
+                    .get_value(&poly_ref)
+                    .ok_or(IncompleteCause::DataNotYetAvailable)?)
+            }
+            PolynomialType::Constant => {
+                let values = self.fixed_data.fixed_cols[&poly_id].values;
+                let row = rows.current_row_index as usize + next as usize;
+                Ok(values[row % values.len()])
```

Comment:
I'm guessing this doesn't actually support arrays in witgen?

Reply:
What do you mean? This is accessing a row of a fixed column. It works in witgen because this is the prover query part, it has access to anything the prover / witgen has access to.

Reply:
the removed code below used to issue an error that "arrays are not supported", but I don't see how that's not reachable anymore

Reply:
Oh right, it turns out it was easy to support them.

Reply:
ok